### PR TITLE
Normative: Adding Intl.RelativeTimeFormat specification

### PR DESCRIPTION
**Pull request by caiolima**
Adding specification text from https://github.com/tc39/proposal-intl-relative-time to the main spec.

### CODE REVIEW

**Comment by longlho** [comment]
Can this wait for the remaining PRs in https://github.com/tc39/proposal-intl-relative-time to be resolved?

**Comment by caiolima** [comment]
> Can this wait for the remaining PRs in https://github.com/tc39/proposal-intl-relative-time to be resolved?

This PR is blocked until we merge all PRs on proposal's repo. Also, it needs to wait until we get stage 4 status from TC39.

**Comment by longlho** [comment]
thanks for the explanation!

**Comment by leobalter** [comment]
@caiolima any updates since then?

**Comment by littledan** [comment]
Now that Intl.RelativeTimeFormat is at Stage 4, I think this PR is good to land (modulo editor reviews).

**Comment by caiolima** [comment]
> @caiolima any updates since then?

@leobalter just rebased it.

File: spec/relativetimeformat.html [hunk 1]
```diff
@@ -0,0 +1,403 @@
...
+        1. If _options_ is *undefined*, then
+          1. Let _options_ be ObjectCreate(*null*).
+        1. Else,
+          1. Let _options_ be ? ToObject(_options_).
+        1. Let _opt_ be a new Record.
+        1. Let _matcher_ be ? GetOption(_options_, `"localeMatcher"`, `"string"`, «`"lookup"`, `"best fit"`», `"best fit"`).
+        1. Set _opt_.[[LocaleMatcher]] to _matcher_.
+        1. Let _numberingSystem_ be ? GetOption(_options_, `"numberingSystem"`, `"string"`, *undefined*, *undefined*).
+        1. If _numberingSystem_ is not *undefined*, then
+          1. If _numberingSystem_ does not match the `(3*8alphanum) *("-" (3*8alphanum))` sequence, throw a *RangeError* exception.
```
Comment (spectranaut) [inline spec/relativetimeformat.html]: Can we replace this line with the analogous line added in the [locale proposal](https://tc39.es/proposal-intl-locale/#sec-Intl.Locale): 

https://github.com/tc39/proposal-intl-locale/blob/895343957adf133bed9cb3c523a80371e9034d7c/spec.html#L209

I think its much more clear!

File: spec/relativetimeformat.html [hunk 2]
```diff
@@ -0,0 +1,403 @@
...
+      <emu-note>
+        Unicode Technical Standard 35 implicitly describes one locale extension key that is relevant to relative time formatting, "nu" for numbering system.
+      </emu-note>
+
+      <p>
+        The value of the [[LocaleData]] internal slot is implementation defined within the constraints described in <emu-xref href="#sec-internal-slots"></emu-xref> and the following additional constraints:
+      </p>
+
+      <ul>
+        <li>[[LocaleData]][Locale] has ordinary data properties `"second"`, `"minute"`, `"hour"`, `"day"`, `"week"`, `"month"`, `"quarter"`, and `"year"`. Additional property keys may exist with the previous names concatenated with the strings `"-narrow"` or `"-short"`. The values corresponding to these property keys are objects which contain these two categories of properties:
```
Comment (spectranaut) [inline spec/relativetimeformat.html]: We are also now using string notation for properties: https://github.com/tc39/ecma402/pull/400

File: spec/relativetimeformat.html [hunk 3]
```diff
@@ -0,0 +1,403 @@
...
+        <table class="real-table">
+          <thead>
+            <tr>
+              <th>Internal Slot</th>
+              <th>Property</th>
+            </tr>
+          </thead>
+          <tr>
+            <td>[[Locale]]</td>
+            <td>`"locale"`</td>
```
Comment (spectranaut) [inline spec/relativetimeformat.html]: These should also be turned to the string literal format.

**Pull request merged**